test/modules/layers/test_codebook.py

```diff
@@ -4,250 +4,306 @@
-            if i == 1:
-                assert_expected(
-                    emb, self.vq.code_avg[1] / self.vq.code_usage[1], rtol=0, atol=1e-4
-                )
-            # Compare each embedding vector to each encoded vector.
-            # If at least one match, then restart happened.
-            else:
-                assert any(
-                    [
-                        torch.isclose(emb, enc, rtol=0, atol=1e-4).all()
-                        for enc in encoded_flat_noise
-                    ]
-                ), "embedding restarted from encoder output incorrectly"
-
-    def test_load_state_dict(self):
-        state_dict = OrderedDict(
-            [
-                ("linear.weight", tensor([[1.0]])),
-                ("linear.bias", tensor([2.0])),
-                ("codebook.embedding", tensor([[3.0]])),
-                ("codebook.code_usage", tensor([4.0])),
-                ("codebook.code_avg", tensor([[5.0]])),
-            ]
-        )
-
-        class DummyModel(nn.Module):
-            def __init__(self):
-                super().__init__()
-                self.linear = nn.Linear(1, 1)
-                self.codebook = Codebook(1, 1)
-
-        model = DummyModel()
-        assert not model.codebook._is_embedding_init
-        model.load_state_dict(state_dict)
-        assert model.codebook._is_embedding_init
-
-        actual = model.codebook.embedding
-        expected = state_dict["codebook.embedding"]
-        assert_expected(actual, expected)
-
-        actual = model.codebook.code_usage
-        expected = state_dict["codebook.code_usage"]
-        assert_expected(actual, expected)
-
-        actual = model.codebook.code_avg
-        expected = state_dict["codebook.code_avg"]
-        assert_expected(actual, expected)
+            ), "embedding restarted from encoder output incorrectly"
+
+
+def test_load_state_dict():
+    state_dict = OrderedDict(
+        [
+            ("linear.weight", tensor([[1.0]])),
+            ("linear.bias", tensor([2.0])),
+            ("codebook.embedding", tensor([[3.0]])),
+            ("codebook.code_usage", tensor([4.0])),
+            ("codebook.code_avg", tensor([[5.0]])),
+        ]
+    )
+
+    class DummyModel(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.linear = nn.Linear(1, 1)
+            self.codebook = Codebook(1, 1)
+
+    model = DummyModel()
+    assert not model.codebook._is_embedding_init
+    model.load_state_dict(state_dict)
+    assert model.codebook._is_embedding_init
+
+    actual = model.codebook.embedding
+    expected = state_dict["codebook.embedding"]
+    assert_expected(actual, expected)
+
+    actual = model.codebook.code_usage
+    expected = state_dict["codebook.code_usage"]
+    assert_expected(actual, expected)
+
+    actual = model.codebook.code_avg
+    expected = state_dict["codebook.code_avg"]
+    assert_expected(actual, expected)
```

Comment:
maybe you could replace these three assertion with the `assert_expected_wrapper` utility?

Reply:
Ok. I added value assertion to the wrapper. Will rebase and incorporate. Thanks for the reminder.

Reply:
Actually, `assert_expected_wrapper` is meant for actual being `NamedTuple` but in this block we are dealing with codebook parameters.

I have changed other places wherever applicable.

Reply:
oh yeah I missed that, makes sense